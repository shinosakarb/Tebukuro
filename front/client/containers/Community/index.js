import React from 'react'
import { connect } from 'react-redux'
import { showCommunity } from '../../actions/Community'
import Community from '../../components/Community'

const mapStateToProps = (state, ownProps) => {
  return {
    community: state.Community
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    // onSubmit: (name, description) => {
    //   dispatch(createCommunity(name, description))
    // }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Community)
