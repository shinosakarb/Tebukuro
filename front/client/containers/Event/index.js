import React, { Component }   from 'react'
import { connect }            from 'react-redux'
import { bindActionCreators } from 'redux'
import * as Actions           from '../../actions/Event'
import EventComponent         from '../../components/Event'
import EventModel             from '../../models/Event'
import TicketListComponent    from '../../components/TicketList'

export class Event extends Component {
  componentDidMount() {
    this.props.actions.showEvent(this.props.params.eventId)
  }

  render() {
    return (
      <div>
        event page sample
        <EventComponent event={ this.props.event } />
        <TicketListComponent tickets={ this.props.event.tickets } />
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    event: state.event
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    actions: bindActionCreators(Actions, dispatch) 
  }
}

Event.propTypes = { event: React.PropTypes.instanceOf(EventModel).isRequired }

export default connect(mapStateToProps, mapDispatchToProps)(Event)
