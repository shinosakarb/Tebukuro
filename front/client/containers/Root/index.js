import React from 'react'
import { Provider } from 'react-redux'
import { Router, Route, IndexRoute } from 'react-router'
import App from '../App'
import Top from '../../components/Top'
import CommunityList from '../CommunityList'
import CreateCommunityForm from '../CommunityForm'

export default class Root extends React.Component {
  static propTypes = {
    history: React.PropTypes.object.isRequired,
    store: React.PropTypes.object.isRequired
  }

  render() {
    const { store, history } = this.props
    return (
      <Provider store={store}>
        <Router history={history}>
          <Route path="/" component={App}>
            <IndexRoute component={Top}/>
            <Route path="communities" component={CommunityList}/>
            <Route path="communities/new" component={CreateCommunityForm}/>
          </Route>
        </Router>
      </Provider>
    )
  }
}
