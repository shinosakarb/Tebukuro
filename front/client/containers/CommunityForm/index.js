import React from 'react'
import { connect } from 'react-redux'
import { createCommunity } from '../../actions/Community'
import CommunityForm from '../../components/CommunityForm'

const mapStateToProps = (state, ownProps) => {
  return {
    community: state.Community
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    onSubmit: (params) => {
      dispatch(createCommunity(params))
    }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(CommunityForm)
