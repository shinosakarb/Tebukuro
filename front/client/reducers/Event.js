import { handleActions }    from 'redux-actions'
import Event                from '../models/Event'
import { ticketReducerMap } from './Ticket'
import ConvertCase          from '../utils/ConvertCase'

export const eventInitialState = new Event()

export const eventReducerMap = {
  CREATE_EVENT: {
    next: (state, action) => {
      const event = new Event(ConvertCase.camelKeysOf(action.payload))
      return event.setTickets(
        action.payload.tickets ? action.payload.tickets.map((item) => {
          return ticketReducerMap.CREATE_TICKET.next(
            null,
            {payload: item}
          )
        }) : []
      )
    },
    throw: (state, action) => {
      return new Event({
        errors: action.payload.messages
      })
    }
  }
}

export default handleActions(eventReducerMap, eventInitialState)
