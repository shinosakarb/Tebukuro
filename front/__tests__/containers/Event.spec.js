import React              from 'react'
import { mount, shallow } from 'enzyme'
import ToJson             from 'enzyme-to-json'
import { Event }          from '../../client/containers/Event'
import EventModel         from '../../client/models/Event'
import TicketModel        from '../../client/models/Ticket'

const eventParams = {
  id: 1,
  name: 'test',
  description: 'This event is active in Umeda.',
  eventStartsAt: '2017-03-04T13:00:00.000Z',
  eventEndsAt: '2017-03-04T17:00:00.000Z',
  address: 'Hilton Plaza West 12F',
}

const ticketsParams = [ 
  {
    id:            3,
    name:          'ticketName1',
    cost:          4,
    quantity:      5,
    eventId:       1,
    saleStartsAt: '2017-03-01T09:00:00.000Z',
    saleEndsAt:   '2017-03-01T10:30:00.000Z'
  },
  {
    id:            7,
    name:          'ticketName2',
    cost:          8,
    quantity:      9,
    eventId:       1,
    saleStartsAt: '2017-12-11T13:30:00.000Z',
    saleEndsAt:   '2017-12-11T15:00:00.000Z'
  }
]

let eventModel = new EventModel(eventParams)
const ticketListModel = ticketsParams.map((ticket) => new TicketModel(ticket))
eventModel = eventModel.setTickets(ticketListModel)

const props = {
  event:   eventModel,
  params:  { eventId: eventModel.id },
  actions: { showEvent: jest.fn() }
}

describe('Event container', () => {

  it('should call showEvent in componentDidMount once', () => {
    const wrapper = mount(<Event {...props} />)
    expect(props.actions.showEvent).toHaveBeenCalledTimes(1)
  })

  it('should render self and subcomponents', () => {
    const wrapper = shallow(<Event {...props} />)
    const tree = ToJson(wrapper)
    expect(tree).toMatchSnapshot()
  })
})
