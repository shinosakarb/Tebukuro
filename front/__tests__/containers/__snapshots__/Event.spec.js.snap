// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event container should render self and subcomponents 1`] = `
<div>
  event page sample
  <Event
    event={
      Object {
        "address": "Hilton Plaza West 12F",
        "communityId": 0,
        "description": "This event is active in Umeda.",
        "errors": Array [],
        "eventEndsAt": "2017-03-04T17:00:00.000Z",
        "eventStartsAt": "2017-03-04T13:00:00.000Z",
        "id": 1,
        "name": "test",
        "tickets": Array [
          Object {
            "cost": 4,
            "errors": Array [],
            "eventId": 1,
            "id": 3,
            "name": "ticketName1",
            "quantity": 5,
            "saleEndsAt": "2017-03-01T10:30:00.000Z",
            "saleStartsAt": "2017-03-01T09:00:00.000Z",
          },
          Object {
            "cost": 8,
            "errors": Array [],
            "eventId": 1,
            "id": 7,
            "name": "ticketName2",
            "quantity": 9,
            "saleEndsAt": "2017-12-11T15:00:00.000Z",
            "saleStartsAt": "2017-12-11T13:30:00.000Z",
          },
        ],
      }
    }
  />
  <TicketList
    tickets={
      Array [
        Object {
          "cost": 4,
          "errors": Array [],
          "eventId": 1,
          "id": 3,
          "name": "ticketName1",
          "quantity": 5,
          "saleEndsAt": "2017-03-01T10:30:00.000Z",
          "saleStartsAt": "2017-03-01T09:00:00.000Z",
        },
        Object {
          "cost": 8,
          "errors": Array [],
          "eventId": 1,
          "id": 7,
          "name": "ticketName2",
          "quantity": 9,
          "saleEndsAt": "2017-12-11T15:00:00.000Z",
          "saleStartsAt": "2017-12-11T13:30:00.000Z",
        },
      ]
    }
  />
</div>
`;
